### PR TITLE
New relic backend

### PR DESCRIPTION
**Pull request by mithuna**
## Description
New Relic is the performance monitoring tool we have decided to use for our application. https://newrelic.com/
This code  will install NR to our infrastructure. 
Env variables are adding to env.template. Should copy that to a  .env file.

## Steps to Test
Merge and deploy in dev

## Deploy Notes

### CODE REVIEW

File: provisioners/configure.sh [hunk 1]
```diff
@@ -16,8 +16,14 @@ apt-get -qq update
 apt-get -qq install -y curl htop wget build-essential zip software-properties-common gnupg awscli
 
 echo $PERM_ENV  > /data/www/host.txt
+#preseeding NR
+echo newrelic-php5 newrelic-php5/application-name string $NEW_RELIC_APPLICATION_NAME | debconf-set-selections
+echo newrelic-php5 newrelic-php5/license-key string $NEW_RELIC_LICENSE_KEY | debconf-set-selections
```
Comment (jasonaowen) [inline provisioners/configure.sh:21]: These preseeding lines look good, but the variables are not yet populated. You'll need to pipe the values from GitHub secrets through packer and ansible. See also [my comment about the Sentry values](https://github.com/PermanentOrg/infrastructure/pull/51#discussion_r610280627).

File: .github/workflows/build-images.yml [hunk 2]
```diff
@@ -51,14 +51,15 @@ jobs:
...
           APP_ID: ${{ matrix.environment.app_id }}
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           DEPLOY_AWS_ACCESS_KEY_ID: ${{ secrets[matrix.environment.aws_deploy_key] }}
           DEPLOY_AWS_SECRET_ACCESS_KEY: ${{ secrets[matrix.environment.aws_deploy_secret] }}
-          AWS_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
-          SENTRY_DSN: ${{ secrets.SENTRY_TOKEN }}
-          PERM_ENV: ${{ matrix.environment.perm_env }}
+          NEW_RELIC_LICENSE_KEY: ${{ secrets.NEW_RELIC_LICENSE_KEY }}
```
Comment (jasonaowen) [inline .github/workflows/build-images.yml:58]: This looks good!

You'll also need to set the `NEW_RELIC_APPLICATION_NAME` variable, in the same way as we're doing `APP_ID`.

File: .env.template [hunk 3]
```diff
@@ -57,3 +57,9 @@ export NOTIFICATION_DATABASE_URL="postgresql://user:pass@host/notifications"
 # Firebase credentials for notification-service
 # These credentials are baked into the image to be used by the application
 export NOTIFICATION_FIREBASE_CREDENTIALS='{"format":"minified JSON";"see":"https://github.com/PermanentOrg/notification-service/"}'
+
+# New Relic credentials for Performance monitoring
+# These are configured on GitHub; they are listed here only for the sake of completeness
+# and can be left blank
+export NEW_RELIC_APPLICATION_NAME_DEV=Backend-dev
```
Comment (jasonaowen) [inline .env.template:64]: I just noticed that the variable name has a `DEV` on the end; we don't want to specify the name of the environment in the names of the variables.
```suggestion
export NEW_RELIC_APPLICATION_NAME=Backend-dev
```

**Review by jasonaowen — COMMENTED** [review]
Good progress, @mithuna!

File: .github/workflows/build-images.yml (hunk 2 above)
Comment (jasonaowen) [reply .github/workflows/build-images.yml:61]: And it needs to be piped all the way through, just like the license key!

Comment (mithuna) [reply .github/workflows/build-images.yml:61]: aah ok.. will do that..

**Comment by mithuna** [comment]
@jasonaowen I pushed some changes and  it is ready for review

**Review by jasonaowen — APPROVED** [review]
This looks good to me! Nice work, @mithuna.

I'd like to do one additional bit of testing before we merge, to make sure we set up the GitHub workflow correctly. Can you please run the build images action on this branch - which will make new AMIs in our usual region - and then manually launch an instance off a new AMI to confirm that data is flowing to NR correctly?

**Comment by jasonaowen** [comment]
Mithuna is out, so I'll do the branch build & dev deploy now.

File: .github/workflows/build-images.yml [hunk 4]
```diff
@@ -51,14 +57,16 @@ jobs:
...
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           DEPLOY_AWS_ACCESS_KEY_ID: ${{ secrets[matrix.environment.aws_deploy_key] }}
           DEPLOY_AWS_SECRET_ACCESS_KEY: ${{ secrets[matrix.environment.aws_deploy_secret] }}
-          AWS_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
-          SENTRY_DSN: ${{ secrets.SENTRY_TOKEN }}
-          PERM_ENV: ${{ matrix.environment.perm_env }}
+          NEW_RELIC_LICENSE_KEY: ${{ secrets[matrix.environment.new_relic_license_key] }}
+          NEW_RELIC_APPLICATION_NAME: ${{ secrets[matrix.environment.new_relic_application_name] }}
```
Comment (jasonaowen) [inline .github/workflows/build-images.yml:65]: This isn't a secret, and so ended up not getting populated:
```suggestion
          NEW_RELIC_APPLICATION_NAME: ${{ matrix.environment.new_relic_application_name }}
```

**Comment by jasonaowen** [comment]
With the last change, I was able to successfully build images, and deploy to dev. I will squash & merge.

**Comment by jasonaowen** [comment]
Oh, but the current changes only include the backend, not also the task-runner and cron. Adding.

File: .github/workflows/build-images.yml (hunk 4 above)
Comment (mithuna) [reply .github/workflows/build-images.yml]: ah sorry.. My bad..

**Comment by mithuna** [comment]
@jasonaowen Thanks for pitching in and getting this done.

File: .github/workflows/build-images.yml (hunk 4 above)
Comment (jasonaowen) [reply .github/workflows/build-images.yml]: No worries, this is why we test! :slightly_smiling_face:

**Pull request merged**